src/boot/efi/util.c

```diff
@@ -57,7 +57,7 @@ EFI_STATUS efivar_set_uint_string(const EFI_GUID *vendor, const CHAR16 *name, UI
         assert(vendor);
         assert(name);
 
-        SPrint(str, ELEMENTSOF(str), L"%u", i);
+        ValueToString(str, FALSE, i);
```

Comment:
hmm, this looks wrong. ValueToString() takes a signed integer... but this one is about unsigned

Reply:
This is exactly what `SPrint` does internally. It does not give a fuck about the sign and will pass UINT32/64 to the INT64 arg of `ValueToString`. Which is why `Print(L"%lu", UINTN_MAX)` gives you `-1`.

I know this is messed up. But I think it's better to call this directly rather than indirectly through the Print machinery.

Reply:
please add inline comment about this.